pkg/crypto/hash/hash.go

```diff
@@ -53,7 +53,7 @@ func RipeMD160(data []byte) util.Uint160 {
 	hasher := ripemd160.New()
 	_, _ = hasher.Write(data)
 
-	hash, _ = util.Uint160DecodeBytesBE(hasher.Sum(nil))
+	hasher.Sum(hash[:0])
```

Comment:
Neat!

Reply:
I wonder if the same thing is beneficial for `DoubleSha256`. `Sum256` allocates `digest` internally but this can (?) be done on stack, so I am not sure in this case.

Reply:
We only use `DoubleSha256` in mpt now and it doesn't appear on the radar at all. At the same time `Sum256` returns `[Size]byte`, so it copies its internal buffer which ensures that it won't be accessible after return from it. Compiler should be able to optimize this allocation to be made on the stack.